src/rashdf/plan.py

```diff
@@ -1117,6 +1121,72 @@ def reference_lines_timeseries_output(self) -> xr.Dataset:
         """
         return self.reference_timeseries_output(reftype="lines")
 
+    def observed_timeseries_input(self, vartype: str = "Flow") -> dict:
+        """Return observed timeseries input data for reference lines and points from a HEC-RAS HDF plan file.
+
+        Parameters
+        ----------
+        vartype : str, optional
+            The type of observed data to retrieve. Must be either "Flow" or "Stage".
+            (default: "Flow")
+
+        Returns
+        -------
+        xr.Dataset
+            An xarray Dataset with observed timeseries input data for both reference lines and reference points.
+        """
+        if vartype == "Flow":
+            output_path = self.OBS_FLOW_OUTPUT_PATH
+        elif vartype == "Stage":
+            output_path = self.OBS_STAGE_OUTPUT_PATH
+        else:
+            raise ValueError('vartype must be either "Flow" or "Stage".')
+
+        observed_group = self.get(output_path)
+        if observed_group is None:
+            raise RasPlanHdfError(
+                f"Could not find HDF group at path '{output_path}'."
+                f" Does the Plan HDF file contain reference {vartype} output data?"
+            )
+        if "Attributes" in observed_group.keys():
+            attr_path = observed_group["Attributes"]
+            attrs_df = pd.DataFrame(attr_path[:]).map(convert_ras_hdf_value)
+
+        das = {}
+        for idx, site in enumerate(observed_group.keys()):
+            if site != "Attributes":
+                # Site Ex: 'Ref Point: Grapevine_Lake_RP'
+                site_path = observed_group[site]
+                site_name = site.split(":")[1][1:]  # Grapevine_Lake_RP
+                ref_type = site.split(":")[0]  # Ref Point
+                if ref_type == "Ref Line":
+                    ref_type = "refln"
+                else:
+                    ref_type = "refpt"
+                df = pd.DataFrame(site_path[:]).map(convert_ras_hdf_value)
+                # Ensure the Date index is unique
+                df = df.drop_duplicates(subset="Date")
+                # Package into an 1D xarray DataArray
+                values = df["Value"].values
+                times = df["Date"].values
+                da = xr.DataArray(
+                    values,
+                    name=vartype,
+                    dims=["Date"],
+                    coords={
+                        "Date": times,
```

Comment:
@zherbz last comment, something I should have caught before -- let's call the `Date` dimension `time` instead for consistency with other methods in this library.

Reply:
col now updated from Date to time